mcmc/SampleSummary.h

```diff
@@ -0,0 +1,322 @@
+#pragma once
+
+// C++ includes
+#include <iostream>
+#include <vector>
+
+// ROOT include
+#include "TH1D.h"
+#include "TH2D.h"
+#include "TH3D.h"
+#include "TH2Poly.h"
+#include "THStack.h"
+#include "TStyle.h"
+#include "TCanvas.h"
+#include "TApplication.h"
+
+// MaCh3 includes
+#include "samplePDF/samplePDFBase.h"
+#include "mcmc/StatisticalUtils.h"
+#include "mcmc/MCMCProcessor.h"
+
+// *******************
+/// @brief Class to calculate pvalue produce posterior predictive and many fancy Bayesian stuff
+class SampleSummary {
+// *******************
+  public:
+    /// @brief Constructor
+    SampleSummary(const int n_Samples, const std::string &Filename, samplePDFBase* const sample, const int nSteps);
```

Comment:
Just to check, this all works for the samplePDFFD classes? I guess testing this out for the FD would be a good test to do soon?

Reply:
Well, the main input to this class are throws which are TH2Poly. I added TH2D->TH2Poly conversion.
samplePDFBase* is only used to get stuff like Binning, Nsamples, sample names etc. Sadly these are virtual fucnitons. Once these virtual fucntions are solved it will work with FD as well.

This shoudl be super simple but I didn't want to touch too much FD code. Otherwise this class is quite generic

Reply:
yeah this is fine. It is more of a mental note for me (or somebody) to check all of this at the FD side.